### ISSUE TITLE
xunit render method error avocado --dry-run on arch-linux

### ISSUE THREAD

**Issue by omps** [avocado-framework/avocado#2439]
```
(avocado) [omps@t61 ~]$ avocado run /bin/true --dry-run
JOB ID     : 0000000000000000000000000000000000000000
JOB LOG    : /var/tmp/avocado-dry-run-eudqrtyy/job-2018-02-11T18.24-0000000/job.log
 (1/1) /bin/true: ERROR (0.00 s)
RESULTS    : PASS 0 | ERROR 1 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 0.14 s
JOB HTML   : /var/tmp/avocado-dry-run-eudqrtyy/job-2018-02-11T18.24-0000000/results.html
Error running method "render" of plugin "xunit": ownerDocument
```
While running the same without dry-run doesn't gives render method for xunit.
```
(avocado) [omps@t61 ~]$ avocado run /bin/true 
JOB ID     : a15d7c233211a59ce5b8cc8e057e66a1c51aeb5b
JOB LOG    : /home/omps/avocado/job-results/job-2018-02-11T18.24-a15d7c2/job.log
 (1/1) /bin/true: PASS (0.02 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 0.22 s
JOB HTML   : /home/omps/avocado/job-results/job-2018-02-11T18.24-a15d7c2/results.html

```

**Comment by ldoktor** [comment]
This is quite odd and looks like some differneces in `python2-lxml`. Unfortunately this dispatcher does not allows to log the full traceback. Would you please try applying https://github.com/avocado-framework/avocado/pull/2445 and using `avocado --show all` (or just `avocado --show app,avocado.debug`) and put the traceback here? It should look similar to this:

```
avocado --show app,avocado.debug run passtest.py
JOB ID     : 20efa860344604dc4ccb3fabeb441bd62b6858f7
JOB LOG    : /home/medic/avocado/job-results/job-2018-02-13T10.28-20efa86/job.log
 (1/1) passtest.py:PassTest.test: PASS (0.02 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 0.25 s
JOB HTML   : /home/medic/avocado/job-results/job-2018-02-13T10.28-20efa86/results.html
avocado.debug: 
avocado.debug: Reproduced traceback from: /home/medic/Work/Projekty/avocado/avocado/avocado/core/dispatcher.py:169
avocado.debug: Traceback (most recent call last):
avocado.debug:   File "/home/medic/Work/Projekty/avocado/avocado/avocado/plugins/xunit.py", line 110, in render
avocado.debug:     raise Exception("adf")
avocado.debug: Exception: adf
avocado.debug: 
Error running method "render" of plugin "xunit": adf
```

**Comment by omps** [comment]
@ldoktor 

please find the output below.

```
(avocado) [omps@t61 ~]$ avocado --show app,avocado.debug run /bin/true --dry-run
JOB ID     : 0000000000000000000000000000000000000000
JOB LOG    : /var/tmp/avocado-dry-run-rrscafaa/job-2018-02-14T20.11-0000000/job.log
 (1/1) /bin/true: ERROR (0.00 s)
RESULTS    : PASS 0 | ERROR 1 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 0.15 s
JOB HTML   : /var/tmp/avocado-dry-run-rrscafaa/job-2018-02-14T20.11-0000000/results.html
avocado.debug: 
avocado.debug: Reproduced traceback from: /home/omps/avocado/lib/python3.6/site-packages/avocado/core/dispatcher.py:169
avocado.debug: Traceback (most recent call last):
avocado.debug:   File "/home/omps/avocado/lib/python3.6/site-packages/avocado/plugins/xunit.py", line 117, in render
avocado.debug:     content = self._render(result)
avocado.debug:   File "/home/omps/avocado/lib/python3.6/site-packages/avocado/plugins/xunit.py", line 103, in _render
avocado.debug:     'error')
avocado.debug:   File "/home/omps/avocado/lib/python3.6/site-packages/avocado/plugins/xunit.py", line 58, in _create_failure_or_error
avocado.debug:     element.setAttribute('type', self._get_attr(test, 'fail_class'))
avocado.debug:   File "/usr/lib64/python3.6/xml/dom/minidom.py", line 741, in setAttribute
avocado.debug:     attr.ownerDocument = self.ownerDocument
avocado.debug: AttributeError: ownerDocument
avocado.debug: 
Error running method "render" of plugin "xunit": ownerDocument

```

**Comment by clebergnu** [comment]
@omps this is because the Python 3 port is still not complete.  It runs, but still has bugs.  That one has already been fixed in my development branch.  The fix is:

```diff
commit ea85510847fa580243dd5a93d7ac598b222c3f57
Author: Cleber Rosa <crosa@redhat.com>
Date:   Tue Feb 13 12:24:30 2018 -0500

    xunit plugin: fix DOM usage
    
    On Python 3, it's not possible to create Elements detached from a
    Document.  The following code fails (later than it should) on Python
    3:
    
       >>> from xml.dom.minidom import Element
       >>> e = Element('foo')
       >>> e.setAttribute('bar', 'baz')
       Traceback (most recent call last):
       File "<stdin>", line 1, in <module>
       File "/usr/lib64/python3.6/xml/dom/minidom.py", line 741, in setAttribute
          attr.ownerDocument = self.ownerDocument
       AttributeError: ownerDocument
    
    Let's create elements within the document, with the `createElement`
    API.  According to the docs "The element is not inserted into the
    document when it is created", so the (unchanged) code that actually puts
    the element in the right place still applies.
    
    Signed-off-by: Cleber Rosa <crosa@redhat.com>

diff --git a/avocado/plugins/xunit.py b/avocado/plugins/xunit.py
index 59e54614..00a344d3 100644
--- a/avocado/plugins/xunit.py
+++ b/avocado/plugins/xunit.py
@@ -54,13 +54,13 @@ class XUnitResult(Result):
         return testcase
 
     def _create_failure_or_error(self, document, test, element_type):
-        element = Element(element_type)
+        element = document.createElement(element_type)
         element.setAttribute('type', self._get_attr(test, 'fail_class'))
         element.setAttribute('message', self._get_attr(test, 'fail_reason'))
         traceback_content = self._escape_cdata(test.get('traceback', self.UNKNOWN))
         traceback = document.createCDATASection(traceback_content)
         element.appendChild(traceback)
-        system_out = Element('system-out')
+        system_out = document.createElement('system-out')
         try:
             with open(test.get("logfile"), "r") as logfile_obj:
                 text_output = logfile_obj.read()
@@ -88,7 +88,7 @@ class XUnitResult(Result):
             if status in ('PASS', 'WARN'):
                 pass
             elif status == 'SKIP':
-                testcase.appendChild(Element('skipped'))
+                testcase.appendChild(document.createElement('skipped'))
             elif status == 'FAIL':
                 element, system_out = self._create_failure_or_error(document,
                                                                     test,

```

**Comment by omps** [comment]
thanks @clebergnu its so obvious but somehow in the flow missed to acknowledge that i am using python 3.
Will apply your patch and see if that works. thanks a lot.

**Comment by ldoktor** [comment]
no problem, (mainly) Cleber is working hard on making py3 fully supported. This is why I like bugzilla, which has a nice template to include description, package versions, steps to reproduce and additional info. Anyway at least we identified a place where we lack tracebacks as well as (hopefully) the main issue.

**Comment by clebergnu** [comment]
This should have been fixed by bfab1c6b20f8489c343cfc8c8964d8905b0f72a1.
